game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts

```diff
@@ -105,11 +112,22 @@ const start = (complete: () => void) => {
     })
 }
 
+function orderFilter(event: ExecuteOrderFilterEvent): boolean {
+    // Only allow to train the allowed ability if set.
+    if (learnAbilityAllowedName && event.order_type === UnitOrder.TRAIN_ABILITY) {
+        const ability = getOrError(EntIndexToHScript(event.entindex_ability), "Could not find ability being trained") as CDOTABaseAbility
+        displayDotaErrorMessage("Train the ability you are instructed to.")
```

Comment:
I would change the error to properly display the name of the required ability.
Also, the error message should be localized, including the name of the ability.

Also, is this showing the error regardless of whether the trained ability's is the right ability, since the check is actually done in the following line?

Reply:
This would be hard to localize

Reply:
> Also, is this showing the error regardless of whether the trained ability's is the right ability, since the check is actually done in the following line?
Thanks, looks like I was very tired when I made this.